### PR TITLE
code restructure, naming changes for readability, min poll rate 10ms

### PR DESCRIPTION
**Pull request by ehahnmtm**
changed naming from do to di, because it's more logical in robotstudio
added a minimum time to read data from plc (10ms)
some name changes

### CODE REVIEW

File: RSConnectDIOToSnap7/CodeBehind.cs [hunk 1]
```diff
@@ -52,21 +52,21 @@ public class CodeBehind : SmartComponentCodeBehind
 		private S7Client client;
 		private bool bCanConnect = false;
 		private bool bPLC_AddrIsValid = false;
-		private bool[] bDI_AddressIsValid = { };
-		private bool[] bDO_AddressIsValid = { };
+        private bool[][] bDIO_AddressIsValid = new bool[2][];
+        public enum IO { Input,Output };
+        private double timePassed = 0;
```
Comment (DenisFR) [inline RSConnectDIOToSnap7/CodeBehind.cs:58]: Please, respect indent.

Comment (ehahnmtm) [reply RSConnectDIOToSnap7/CodeBehind.cs:58]: hmm, I don't know what happened to the intendation, i take a look on it

File: RSConnectDIOToSnap7/Sharp7.cs [hunk 2]
```diff
@@ -335,8 +335,10 @@ public void Close()
 
         private void CreateSocket()
         {
-            TCPSocket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
-            TCPSocket.NoDelay = true;
+            TCPSocket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp)
+            {
+                NoDelay = true
+            };
```
Comment (DenisFR) [inline RSConnectDIOToSnap7/Sharp7.cs]: Don't modify this file. It is official one.
At least, you can update it by the file you can found here:
https://sourceforge.net/projects/snap7/files/Sharp7/Last/1.0.6/

File: RSConnectDIOToSnap7/CodeBehind.cs (hunk 1 above)
Comment (DenisFR) [reply RSConnectDIOToSnap7/CodeBehind.cs:58]: The best is to show white space characters.

File: RSConnectDIOToSnap7/CodeBehind.cs [hunk 3]
```diff
@@ -52,21 +52,21 @@ public class CodeBehind : SmartComponentCodeBehind
 		private S7Client client;
 		private bool bCanConnect = false;
 		private bool bPLC_AddrIsValid = false;
-		private bool[] bDI_AddressIsValid = { };
-		private bool[] bDO_AddressIsValid = { };
+        private bool[][] bDIO_AddressIsValid = new bool[2][];
+        public enum IO { Input,Output };
```
Comment (DenisFR) [inline RSConnectDIOToSnap7/CodeBehind.cs]: Best to force enum int value:
public enum IO { Input=0, Output=1 };

File: RSConnectDIOToSnap7/CodeBehind.cs [hunk 4]
```diff
@@ -52,21 +52,21 @@ public class CodeBehind : SmartComponentCodeBehind
 		private S7Client client;
 		private bool bCanConnect = false;
 		private bool bPLC_AddrIsValid = false;
-		private bool[] bDI_AddressIsValid = { };
-		private bool[] bDO_AddressIsValid = { };
+        private bool[][] bDIO_AddressIsValid = new bool[2][];
```
Comment (DenisFR) [inline RSConnectDIOToSnap7/CodeBehind.cs]: I prefer you use IO enum size to know 2 is it:
private bool[][] bDIO_AddressIsValid = new bool[((IO[])Enum.GetValues(typeof(IO))).Distinct().Count()][];
IO need to be define before and to add
using System.Linq;
after System.Collections.Generic;

File: RSConnectDIOToSnap7/CodeBehind.cs (hunk 1 above)
Comment (ehahnmtm) [reply RSConnectDIOToSnap7/CodeBehind.cs:58]: in my version, the indentation is fine. maybe this is just a visual from github?

Comment (ehahnmtm) [reply RSConnectDIOToSnap7/CodeBehind.cs:58]: there is also an automatic formatting function in visual studio, under edit->advanced->format document or selection. i give it a try.

File: RSConnectDIOToSnap7/CodeBehind.cs (hunk 4 above)
Comment (ehahnmtm) [reply RSConnectDIOToSnap7/CodeBehind.cs]: hmm, this is not the best for readability, but i also dont know better at the moment

File: RSConnectDIOToSnap7/CodeBehind.cs [hunk 5]
```diff
@@ -76,13 +76,17 @@ public override void OnInitialize(SmartComponent component)
 		public override void OnLoad(SmartComponent component)
 		{
 			base.OnLoad(component);
-			CheckClientAndValues(component);
+
+            bDIO_AddressIsValid[0] = new bool[0];
```
Comment (DenisFR) [inline RSConnectDIOToSnap7/CodeBehind.cs]: Prefer use:
bDIO_AddressIsValid[(int)IO.Output]

File: RSConnectDIOToSnap7/RSConnectDIOToSnap7.csproj [hunk 6]
```diff
@@ -41,12 +41,12 @@
   <ItemGroup>
     <Reference Include="ABB.Robotics.RobotStudio">
       <SpecificVersion>False</SpecificVersion>
-      <HintPath>C:\Program Files (x86)\ABB Industrial IT\Robotics IT\SDK\RobotStudio SDK 6.07\ABB.Robotics.RobotStudio.dll</HintPath>
+      <HintPath>C:\Program Files (x86)\ABB Industrial IT\Robotics IT\SDK\RobotStudio SDK 6.08\ABB.Robotics.RobotStudio.dll</HintPath>
```
Comment (DenisFR) [inline RSConnectDIOToSnap7/RSConnectDIOToSnap7.csproj:44]: Could you update lines 26 and 36 too?

File: RSConnectDIOToSnap7/RSConnectDIOToSnap7.csproj [hunk 7]
```diff
@@ -77,6 +77,7 @@
   -->
   <PropertyGroup>
     <PostBuildEvent>copy /y "$(TargetPath)" "$(ProjectDir)"
-"C:\Program Files (x86)\ABB Industrial IT\Robotics IT\RobotStudio 6.07\bin\LibraryCompiler.exe" "$(ProjectDir)\RSConnectDIOToSnap7.xml"</PostBuildEvent>
+
```
Comment (DenisFR) [inline RSConnectDIOToSnap7/RSConnectDIOToSnap7.csproj:80]: Remove this blank line.

File: RSConnectDIOToSnap7/RSConnectDIOToSnap7.en.xml [hunk 8]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <LibraryResource xmlns="urn:abb-robotics-robotstudio-libraryresource"
                  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-                 xsi:schemaLocation="urn:abb-robotics-robotstudio-libraryresource file:///C:\Program%20Files%20(x86)\ABB%20Industrial%20IT\Robotics%20IT\SDK\RobotStudio%20SDK%206.07\LibraryResourceSchema.xsd">
+                 xsi:schemaLocation="urn:abb-robotics-robotstudio-libraryresource file:///C:\Program%20Files%20(x86)\ABB%20Industrial%20IT\Robotics%20IT\SDK\RobotStudio%20SDK%206.08\LibraryResourceSchema.xsd">
 	<SmartComponent name="RSConnectDIOToSnap7" description="Allow to connect Robot with real or simulated S7 PLC using Sharp7(Snap7).">
 		<DynamicProperty name="Status" description="The status of this component."/>
 		<DynamicProperty name="PLC_Addr" description="IP Address of PLC IPV4.
```
Comment (DenisFR) [inline RSConnectDIOToSnap7/RSConnectDIOToSnap7.en.xml:7]: Could you change description of Dx_Number to "Number of PLC Dx to connect to in Station Logic..."

**Pull request closed without merging**